extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentKiller.java

```diff
@@ -55,6 +72,16 @@ public void kill(DataSegment segment) throws SegmentLoadingException
     }
   }
 
+  private boolean safeNonRecursiveDelete(FileSystem fs, Path path)
+  {
+    try {
+      return fs.delete(path, false);
+    }
+    catch (Exception ex) {
+      return false;
```

Comment:
not really, this is used to delete version, interval etc directories and it is legit for those deletions to fail as they might not be empty. FileSystem documentation about exactly what cases throw exception is not that clear so it is probably ok for failure be considered valid.